File: 0501~1000/0991. Broken Calculator/main.py
"""
Source: https://rb.gy/ncb01
Date: 2023/7/26
Skill:
Ref:
Runtime: 1178 ms, faster than 49.29%
Memory Usage: 30.42 MB, less than 62.61%
Time complexity:
Space complexity:
Constraints:

"""

import math, sys
from typing import List, Optional
from collections import defaultdict, Counter, deque
from heapq import heapify, heappush, heappop, nsmallest
import heapq
import functools
from bisect import bisect_left, bisect_right


class Solution:
    def brokenCalc(self, startValue: int, target: int) -> int:
        if startValue >= target:
            return startValue - target
        elif startValue * 2 > target:


if __name__ == "__main__":
    s = Solution()
    startValue, target = 3, 10
    res = s.brokenCalc(startValue, target)

File: 0501~1000/0991. Broken Calculator/main.cpp
/**
 * Source: t.ly/Kvjwa
 * Date: 2023/11/14
 * Skill:
 * Runtime: 0 ms, faster than 100.00% of C++ online submissions
 * Memory Usage: 6.24 MB, less than 68.41% of C++ online submissions
 * Time complexity: O(n)
 * Space complexity: O(n)
 * Constraints:
 *
 */


#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
#include <set>
#include <unordered_set>
#include <queue>
#include <numeric>
#include <tuple>
#include <stack>
#include <bitset>
#include <functional>
#include <list>


using namespace std;


class Solution {
public:
    int brokenCalc(int startValue, int target) {
        if (startValue >= target) return startValue - target;
        else if (target % 2 == 1) return brokenCalc(startValue, target + 1) + 1;
        else return brokenCalc(startValue, target / 2) + 1;
    }
};


int main() {
    Solution s;
}